jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ScanningAppProvider.java

```diff
@@ -226,7 +226,7 @@ protected void doStart() throws Exception
         List<Path> files = new ArrayList<>();
         for (Resource resource : _monitored)
         {
-            if (resource.exists() && Files.isReadable(resource.getPath()))
+            if (resource != null && Files.isReadable(resource.getPath()))
```

Comment:
So `resource != null` doesn't ensure the file exists, but then Files.isReadable() does an existence check, so that is OK.  But `getPath` can return null, so really this should be:

```suggestion
            if (resource != null && resource.getPath() != null && Files.isReadable(resource.getPath()))
```

Moreover, there is nothing stopping the monitored resource being a composite, so it may definitely have a null Path, but we probably do need to check directories for readability.  Thus it still would be better to just be:

```suggestion
            if (resource != null && resource.isReadable())
```

Reply:
Using new `Resources` utility class instead